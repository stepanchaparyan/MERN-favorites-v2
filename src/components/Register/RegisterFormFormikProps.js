import * as Yup from 'yup';

const initialValues = {
  name: '',
  email: '',
  password: '',
  password2: '',
};

const validationSchema = Yup.object({
  name: Yup.string().required('Required'),
  email: Yup.string()
    .email('Invalid Email format')
    .required('Required'),
  password: Yup.string()
    .min(6, 'Password must have min 6 symbols')
    .required('Required'),
  password2: Yup.string()
    .oneOf([Yup.ref('password'), null], 'Passwords must match')
    .required('Required'),
});

const formikProps = () => ({
  initialValues,
  validateOnChange: true,
  validationSchema,
});

export default formikProps;
