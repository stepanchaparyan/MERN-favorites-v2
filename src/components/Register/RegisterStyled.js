import styled from 'styled-components';
import { tabletUp } from '../../styles/mediaQueries/mixins';
import { Form, Field } from 'formik';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  color: ${props => props.theme.cadetblue};
`;

export const Title = styled.div`
  display: flex;
  justify-content: center;
  padding: 64px 32px 32px;
  font-weight: 600;
  font-size: 30px;
`;

export const FormStyled = styled(Form)`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const FieldStyled = styled(Field)`
  border: 0px;
  border-bottom: 1px solid ${props => props.theme.cadetblue};
  width: 50%;
  padding: 7px 10px;
  margin: 2px;
  outline: none;
  ${tabletUp`
    width: 30%;
  `};
`;

export const RegisterButton = styled.button`
  border-bottom: 1px solid ${props => props.theme.cadetblue};
  padding: 7px 10px;
  margin: 2px;
  outline: none;
  width: 40%;
  border: 0;
  background-color: ${props => props.theme.cadetblue};
  font-weight: 600;
  font-size: 16px;
  margin-top: 16px;
  border-radius: 4px;
  cursor: pointer;
  ${tabletUp`
    width: 20%;
  `};
`;

export const Errors = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px 0 0;
`;

export const QuestionText = styled.div`
  display: flex;
  align-self: center;
  margin: 10px;
  & > a {
    text-decoration: none;
    color: black;
    font-weight: 600;
  }
`;

export const ErrorMessages = styled.div`
  color: red;
  width: 30%;
  font-size: 12px;
`;

export const ErrorButton = styled.button`
  color: red;
  background-color: transparent;
  border: none;
  outline: none;
  cursor: pointer;
`;

export const modalStyles = {
  content: {
    border: 'none',
    position: 'unset'
  },
  overlay: {
    position: 'fixed',
    inset: '0%',
    zIndex: '1'
  }
};

export const ModalContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: auto;
  padding: 32px;
  text-align: center;
  ${tabletUp`
    max-width: 500px;
    margin: auto;
    padding: 100px;
  `};
`;

export const ButtonConfirm = styled.button`
  border: 0;
  border-radius: 4px;
  background-color: ${props => props.theme.cadetblue};
  font-weight: 600;
  font-size: 16px;
  cursor: pointer;
  margin: auto;
  padding: 8px;
  min-width: 120px;
  width: fit-content;
`;

export const ModalTitleContainer = styled.div`
  background-color: ${props => props.theme.cadetblue};
  display: flex;
  justify-content: center;
`;

export const ModalTitle = styled.div`
  font-size: 20px;
  padding: 16px;
  font-weight: bold;
`;

export const ModalTextContainer = styled.div`
  display: flex;
  min-height: 80px;
  justify-content: center;
  color: ${props => props.theme.cadetblue};
  font-size: 18px;
  font-weight: bold;
  align-items: center;
  padding: 20px;
  text-align: center;
  white-space: pre-wrap;
`;
