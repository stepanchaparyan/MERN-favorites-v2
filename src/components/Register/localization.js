import { defineMessages } from 'react-intl';

export default defineMessages({
  signup: {
    id: 'register.signup',
    defaultMessage: 'Sign Up',
  },
  name: {
    id: 'register.name',
    defaultMessage: 'Name',
  },
  email: {
    id: 'register.email',
    defaultMessage: 'Email',
  },
  password: {
    id: 'register.password',
    defaultMessage: 'Password',
  },
  confirmPassword: {
    id: 'register.confirmPassword',
    defaultMessage: 'Confirm Password',
  },
  signin: {
    id: 'register.signin',
    defaultMessage: 'Sign In',
  },
  alreadyHaveAnAccout: {
    id: 'register.alreadyHaveAnAccout',
    defaultMessage: 'Already have an accout?',
  },
  modalTitle: {
    id: 'addProfile.modalTitle',
    defaultMessage: 'Welcome to our board',
  },
  modalText: {
    id: 'addProfile.modalText',
    defaultMessage: 'There you can create cards for your favorite Items, favorite products and add them to your card',
  },
  modalButtonConfirmText: {
    id: 'addProfile.modalButtonConfirmText',
    // eslint-disable-next-line quotes
    defaultMessage: "Let's go",
  },
});
