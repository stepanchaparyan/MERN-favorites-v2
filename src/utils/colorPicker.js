import theme from '../styles/theme';
import { FAVITEM } from '../constants';

const { red, blue, limeGreen, orange, black } = theme;
const { TYPE } = FAVITEM;

export const colorPicker = category => {
  let color;
  switch (category) {
    case TYPE.FILM:
      color = red;
      break;
    case TYPE.MUSIC:
      color = blue;
      break;
    case TYPE.BOOKS:
      color = limeGreen;
      break;
    case TYPE.OTHER:
      color = orange;
      break;
    default:
      color = black;
  }
  return color;
};
