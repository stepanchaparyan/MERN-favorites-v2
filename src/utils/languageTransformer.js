const dict = {
  // lang - country
  ar: 'sa',
  be: 'bl',
  cs: 'sz',
  da: 'dk',
  de: 'de',
  el: 'gr',
  en: 'us',
  es: 'es',
  fa: 'ir',
  fr: 'fr',
  ja: 'jp',
  hi: 'in',
  hu: 'hu',
  hr: 'hr',
  hy: 'am',
  it: 'it',
  kk: 'kz',
  nl: 'nl',
  no: 'no',
  pl: 'pl',
  pt: 'pt',
  ru: 'ru',
  sk: 'sk',
  sl: 'sl',
  sv: 'se',
  tr: 'tr',
  uk: 'ua'
};

export const languageTransformer = listList => {
  let newList = [];
  listList.map(lang => {
    const item = Object.keys(dict).find(item => item === lang);
    if (item) {
      newList.push(dict[item].toUpperCase());
    }
  });
  return newList;
};

function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export const setLanguage = (code, setLocale) => {
  const countryCode = Object.values(dict).includes(code.toLowerCase());

  if (countryCode) {
    setLocale(getKeyByValue(dict, code.toLowerCase()));
  }
};
