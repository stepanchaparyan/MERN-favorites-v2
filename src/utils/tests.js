import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
