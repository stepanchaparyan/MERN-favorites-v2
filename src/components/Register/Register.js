import React, { useState, useContext, useEffect, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { Link } from 'react-router-dom';
import { Formik, ErrorMessage } from 'formik';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import AuthContext from '../../context/authContext/authContext';
import ProfileContext from '../../context/profileContext/profileContext';
import {
  Container,
  Title,
  RegisterButton,
  QuestionText,
  FormStyled,
  FieldStyled,
  ErrorMessages,
  Errors,
  ErrorButton,
  modalStyles,
  ModalContainer,
  ModalTitleContainer,
  ModalTitle,
  ModalTextContainer,
  ButtonConfirm,
} from './RegisterStyled';
import localization from './localization';
import { FORM, LINK } from '../../constants';
import registerFormFormikProps from './RegisterFormFormikProps';
import Loading from '../../components/Loading/Loading';

const { INPUT } = FORM;

const Register = props => {
  const { register, user, isAuthencated, error, clearErrors, loading } = useContext(AuthContext);
  const { formatMessage } = useIntl();
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const { addProfile } = useContext(ProfileContext);

  useEffect(() => {
    if (isAuthencated && !modalIsOpen) {
      props.history.push('/');
    }
  }, [isAuthencated, props.history]);

  const onSubmit = ({ name, email, password, password2 }) => {
    if (password === password2) {
      register({
        name,
        email,
        password,
      });
      setModalIsOpen(true);
    }
  };

  const onConfirm = () => {
    addProfile({ name: user.name, email: user.email });
    setModalIsOpen(false);
    props.history.push('/');
  };

  const { validationSchema, initialValues } = useMemo(() => registerFormFormikProps(), []);

  return (
    <Container>
      <Title>{formatMessage(localization.signup)}</Title>
      <Modal isOpen={modalIsOpen} onConfirm={onConfirm} style={modalStyles}>
        <ModalContainer>
          <ModalTitleContainer>
            <ModalTitle>{formatMessage(localization.modalTitle)}</ModalTitle>
          </ModalTitleContainer>
          <ModalTextContainer>{formatMessage(localization.modalText)}</ModalTextContainer>
          <ButtonConfirm onClick={onConfirm}>{formatMessage(localization.modalButtonConfirmText)}</ButtonConfirm>
        </ModalContainer>
      </Modal>
      {loading ? (
        <Loading type='ThreeDots' color='cadetblue' height={20} width={60}></Loading>
      ) : (
        !modalIsOpen && (
          <>
            <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onSubmit}>
              <FormStyled>
                <FieldStyled type={INPUT.TYPE.TEXT} name={INPUT.NAME.NAME} id={INPUT.NAME.NAME} placeholder={formatMessage(localization.name)} />
                <ErrorMessage name={INPUT.NAME.NAME} component={ErrorMessages} />
                <FieldStyled type={INPUT.TYPE.EMAIL} name={INPUT.NAME.EMAIL} id={INPUT.NAME.EMAIL} placeholder={formatMessage(localization.email)} />
                <ErrorMessage name={INPUT.NAME.EMAIL} component={ErrorMessages} />
                <FieldStyled
                  type={INPUT.TYPE.PASSWORD}
                  name={INPUT.NAME.PASSWORD}
                  id={INPUT.NAME.PASSWORD}
                  placeholder={formatMessage(localization.password)}
                />
                <ErrorMessage name={INPUT.NAME.PASSWORD} component={ErrorMessages} />
                <FieldStyled
                  type={INPUT.TYPE.PASSWORD}
                  name={INPUT.NAME.PASSWORD_2}
                  id={INPUT.NAME.PASSWORD_2}
                  placeholder={formatMessage(localization.confirmPassword)}
                />
                <ErrorMessage name={INPUT.NAME.PASSWORD_2} component={ErrorMessages} />
                {error && (
                  <Errors>
                    <ErrorButton type={INPUT.TYPE.BUTTON} onClick={() => clearErrors()}>
                      {error}
                    </ErrorButton>
                  </Errors>
                )}
                <RegisterButton type={INPUT.TYPE.SUBMIT}>{formatMessage(localization.signup)}</RegisterButton>
              </FormStyled>
            </Formik>
            <QuestionText>
              {formatMessage(localization.alreadyHaveAnAccout)} &nbsp;
              <Link to={LINK.TO.LOGIN}>{formatMessage(localization.signin)}</Link>
            </QuestionText>
          </>
        )
      )}
    </Container>
  );
};

Register.propTypes = {
  history: PropTypes.object,
};

export default Register;
