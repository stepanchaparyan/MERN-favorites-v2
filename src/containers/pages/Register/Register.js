import React, { useContext, useEffect, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { Link } from 'react-router-dom';
import { Formik, ErrorMessage } from 'formik';
import PropTypes from 'prop-types';
import AuthContext from '../../../context/authContext/authContext';
import {
  Container,
  Title,
  RegisterButton,
  QuestionText,
  FormStyled,
  FieldStyled,
  ErrorMessages,
  Errors,
  ErrorButton
} from './RegisterStyled';
import localization from './localization';
import { FORM, LINK } from '../../../constants';
import registerFormFormikProps from './RegisterFormFormikProps';
import Loading from '../../../components/Loading/Loading';

const { INPUT } = FORM;

const Register = props => {
  const { register, isAuthencated, error, clearErrors, loading } = useContext(AuthContext);
  const { formatMessage } = useIntl();

  useEffect(() => {
    if (isAuthencated) {
      props.history.push('/');
    }
  }, [isAuthencated, props.history]);

  const onSubmit = ({ name, email, password, password2 }) => {
    if (password === password2) {
      register({
        name,
        email,
        password
      });
    }
  };

  const { validationSchema, initialValues } = useMemo(() => registerFormFormikProps(), []);

  return (
    <Container>
      <Title>{formatMessage(localization.signup)}</Title>
      {loading ? (
        <Loading type="ThreeDots" color="cadetblue" height={20} width={60}></Loading>
      ) : (
        <>
          <Formik
            initialValues={initialValues}
            validationSchema={validationSchema}
            onSubmit={onSubmit}
          >
            <FormStyled>
              <FieldStyled
                type={INPUT.TYPE.TEXT}
                name={INPUT.NAME.NAME}
                id={INPUT.NAME.NAME}
                placeholder={formatMessage(localization.name)}
              />
              <ErrorMessage name={INPUT.NAME.NAME} component={ErrorMessages} />
              <FieldStyled
                type={INPUT.TYPE.EMAIL}
                name={INPUT.NAME.EMAIL}
                id={INPUT.NAME.EMAIL}
                placeholder={formatMessage(localization.email)}
              />
              <ErrorMessage name={INPUT.NAME.EMAIL} component={ErrorMessages} />
              <FieldStyled
                type={INPUT.TYPE.PASSWORD}
                name={INPUT.NAME.PASSWORD}
                id={INPUT.NAME.PASSWORD}
                placeholder={formatMessage(localization.password)}
              />
              <ErrorMessage name={INPUT.NAME.PASSWORD} component={ErrorMessages} />
              <FieldStyled
                type={INPUT.TYPE.PASSWORD}
                name={INPUT.NAME.PASSWORD_2}
                id={INPUT.NAME.PASSWORD_2}
                placeholder={formatMessage(localization.confirmPassword)}
              />
              <ErrorMessage name={INPUT.NAME.PASSWORD_2} component={ErrorMessages} />
              {error && (
                <Errors>
                  <ErrorButton type={INPUT.TYPE.BUTTON} onClick={() => clearErrors()}>
                    {error}
                  </ErrorButton>
                </Errors>
              )}
              <RegisterButton type={INPUT.TYPE.SUBMIT}>
                {formatMessage(localization.signup)}
              </RegisterButton>
            </FormStyled>
          </Formik>
          <QuestionText>
            {formatMessage(localization.alreadyHaveAnAccout)} &nbsp;
            <Link to={LINK.TO.LOGIN}>{formatMessage(localization.signin)}</Link>
          </QuestionText>
        </>
      )}
    </Container>
  );
};

Register.propTypes = {
  history: PropTypes.object
};

export default Register;
