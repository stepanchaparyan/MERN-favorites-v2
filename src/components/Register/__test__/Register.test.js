import React, { useContext } from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { renderSnapshot } from '../../../utils/tests';
import Register from '../Register.js';
import { FieldStyled, FormStyled, ErrorButton } from '../RegisterStyled';

jest.mock('react-intl', () => ({
  useIntl: jest.fn(() => ({ formatMessage: m => m.defaultMessage })),
  defineMessages: i => i
}));

jest.mock('react', () => {
  const ActualReact = jest.requireActual('react');
  return {
    ...ActualReact,
    useEffect: jest.fn().mockImplementation(f => f()),
    useContext: jest.fn().mockImplementation(() => ({
      isAuthencated: false,
      error: 'a',
      clearErrors: jest.fn(),
      register: jest.fn()
    }))
  };
});

describe('Register component test with Enzyme', () => {
  test('renders snapshot', () => {
    renderSnapshot(Register);
  });

  test('should click on Input elements', () => {
    const component = shallow(<Register />);
    const inputs = component.find(FieldStyled);
    inputs.forEach(input => {
      input.simulate('change', { target: { value: 'target' } });
    });
  });

  test('should click on history', () => {
    useContext.mockImplementationOnce(
      jest.fn(() => ({
        isAuthencated: true,
        clearErrors: jest.fn()
      }))
    );

    const historyMock = { push: jest.fn() };
    const component = shallow(<Register history={historyMock} />);
    component.find(FormStyled);
    expect(historyMock.push.mock.calls[0]).toEqual(['/']);
  });

  test('should click on sumbit element', () => {
    const component = shallow(<Register />);
    const form = component.find('Formik');
    form.simulate('submit', { target: { name: 'password', value: '123' } });
  });

  test('should click on errorButton elements', () => {
    const component = shallow(<Register />);
    const errorButton = component.find(ErrorButton);
    errorButton.simulate('click');
  });

  test('should click on errorButton elements with errors list', () => {
    useContext.mockImplementationOnce(
      jest.fn(() => ({
        error: ['a', 'b'],
        clearErrors: jest.fn()
      }))
    );

    const component = shallow(<Register />);
    const errorButton = component.find(ErrorButton).at(0);
    errorButton.simulate('click');
  });
});
