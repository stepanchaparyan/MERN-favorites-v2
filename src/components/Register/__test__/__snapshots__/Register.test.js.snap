// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component test with Enzyme renders snapshot 1`] = `
<styled.div>
  <styled.div>
    Sign Up
  </styled.div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentElement={[Function]}
    isOpen={false}
    onConfirm={[Function]}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "border": "none",
          "position": "unset",
        },
        "overlay": Object {
          "inset": "0%",
          "position": "fixed",
          "zIndex": "1",
        },
      }
    }
  >
    <styled.div>
      <styled.div>
        <styled.div>
          Welcome to our board
        </styled.div>
      </styled.div>
      <styled.div>
        There you can create cards for your favorite Items, favorite products and add them to your card
      </styled.div>
      <styled.button
        onClick={[Function]}
      >
        Let's go
      </styled.button>
    </styled.div>
  </Modal>
  <Formik
    initialValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "password2": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "_excludedEdges": Array [],
        "_mutate": undefined,
        "_nodes": Array [
          "password2",
          "password",
          "email",
          "name",
        ],
        "_sortErrors": [Function],
        "_typeError": [Function],
        "_whitelist": ReferenceSet {
          "list": Set {},
          "refs": Map {},
        },
        "conditions": Array [],
        "deps": Array [],
        "exclusiveTests": Object {},
        "fields": Object {
          "email": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "email": false,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "name": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "password": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_whitelistError": undefined,
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "min": true,
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "password2": StringSchema {
            "_blacklist": ReferenceSet {
              "list": Set {},
              "refs": Map {},
            },
            "_blacklistError": undefined,
            "_mutate": undefined,
            "_typeError": [Function],
            "_whitelist": ReferenceSet {
              "list": Set {
                null,
              },
              "refs": Map {
                "password" => Reference {
                  "getter": [Function],
                  "isContext": false,
                  "isSibling": true,
                  "isValue": false,
                  "key": "password",
                  "map": undefined,
                  "path": "password",
                },
              },
            },
            "_whitelistError": [Function],
            "conditions": Array [],
            "deps": Array [],
            "exclusiveTests": Object {
              "required": true,
            },
            "spec": Object {
              "abortEarly": true,
              "nullable": false,
              "presence": "required",
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "spec": Object {
          "abortEarly": true,
          "nullable": false,
          "presence": "optional",
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  >
    <Styled(Form)>
      <Styled(Field)
        id="name"
        name="name"
        placeholder="Name"
        type="text"
      />
      <FormikConnect(ErrorMessageImpl)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 1059715860,
              "baseStyle": undefined,
              "componentId": "sc-iBPRYJ",
              "isStatic": false,
              "rules": Array [
                "color:red;width:30%;font-size:12px;",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-iBPRYJ",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        name="name"
      />
      <Styled(Field)
        id="email"
        name="email"
        placeholder="Email"
        type="email"
      />
      <FormikConnect(ErrorMessageImpl)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 1059715860,
              "baseStyle": undefined,
              "componentId": "sc-iBPRYJ",
              "isStatic": false,
              "rules": Array [
                "color:red;width:30%;font-size:12px;",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-iBPRYJ",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        name="email"
      />
      <Styled(Field)
        id="password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <FormikConnect(ErrorMessageImpl)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 1059715860,
              "baseStyle": undefined,
              "componentId": "sc-iBPRYJ",
              "isStatic": false,
              "rules": Array [
                "color:red;width:30%;font-size:12px;",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-iBPRYJ",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        name="password"
      />
      <Styled(Field)
        id="password2"
        name="password2"
        placeholder="Confirm Password"
        type="password"
      />
      <FormikConnect(ErrorMessageImpl)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 1059715860,
              "baseStyle": undefined,
              "componentId": "sc-iBPRYJ",
              "isStatic": false,
              "rules": Array [
                "color:red;width:30%;font-size:12px;",
              ],
              "staticRulesId": "",
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "sc-iBPRYJ",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        name="password2"
      />
      <styled.div>
        <styled.button
          onClick={[Function]}
          type="button"
        >
          a
        </styled.button>
      </styled.div>
      <styled.button
        type="submit"
      >
        Sign Up
      </styled.button>
    </Styled(Form)>
  </Formik>
  <styled.div>
    Already have an accout?
      
    <Link
      to="/login"
    >
      Sign In
    </Link>
  </styled.div>
</styled.div>
`;
